server/lib/nl/fulfillment/basic.py

```diff
@@ -63,10 +63,13 @@ def populate(state: PopulateState, chart_vars: ChartVars, places: List[Place],
                             [p.dcid for p in places])
     return False
 
-  if chart_vars.source_topic != PROJECTED_TEMP_TOPIC:
+  if (chart_vars.source_topic != PROJECTED_TEMP_TOPIC and
+      not params.is_toolformer_mode(state.uttr.mode)):
```

Comment:
this is very subtle. For easier tracking, can they be consolidated? Or some comment about PROJECTED_TEMP_TOPIC?

I feel it would be nice to have one file with ALL the condition checks, so can find the logic flow in one pass. Maybe something for later.

Reply:
Added some comments.

I'm not sure how to do that, considering there are different fulfillers for different scenarios and fallback across them.  But as a general point, yes, this could use simplification!